src/keystore.c

```diff
@@ -427,6 +427,23 @@ static bool _get_xprv(const uint32_t* keypath, const size_t keypath_len, struct
     return true;
 }
 
+bool keystore_get_root_fingerprint(uint8_t* fingerprint)
+{
+    if (sizeof(fingerprint) < sizeof(uint32_t)) {
```

Comment:
this does not work as expected; `sizeof(singerprint)` is the size of the pointer, which happens to be 4 bytes on the device, but 8 bytes on x86_64. That's why my comment said to either pass the length and check it here, or drop it completely and just describe the required size in the docstring :)

Reply:
I dropped the check entirely now.